File: src/main/java/coza/trojanc/receipt/context/impl/DefaultContextMap.java
package coza.trojanc.receipt.context.impl;

import coza.trojanc.receipt.context.ContextMap;

import java.util.HashMap;
import java.util.Map;


/**
 * Default implementation of a {@link ContextMap}
 * @author Charl Thiem
 */
public class DefaultContextMap implements ContextMap {

	private final Map<String, String> resolvedVariables;

	/**
	 * Instantiates a new Default context map.
	 */
	public DefaultContextMap(){
		this(0);
	}

	/**
	 * Instantiates a new Default context map.
	 *
	 * @param initialSize the initial size
	 */
	public DefaultContextMap(int initialSize){
		resolvedVariables = new HashMap<>(initialSize);
	}

	@Override
	public void add(String name, String value){
		resolvedVariables.put(name, value);
	}

	@Override
	public String get(String name){
		return resolvedVariables.get(name);
	}

	@Override
	public boolean has(String key) {
		return resolvedVariables.containsKey(key);
	}
}

File: src/main/java/coza/trojanc/receipt/context/impl/DefaultContextMap.kt
package coza.trojanc.receipt.context.impl

import coza.trojanc.receipt.context.ContextMap

/**
 * Default implementation of a [ContextMap]
 * @author Charl Thiem
 */
class DefaultContextMap
/**
 * Instantiates a new Default context map.
 *
 * @param initialSize the initial size
 */
@JvmOverloads constructor(initialSize: Int = 0) : ContextMap {

    private val resolvedVariables: MutableMap<String, String>

    init {
        resolvedVariables = hashMapOf();
    }

    override fun add(name: String, value: String) {
        resolvedVariables.put(name, value)
    }

    override fun get(name: String): String? {
        return resolvedVariables.get(name)
    }

    override fun has(key: String): Boolean {
        return resolvedVariables.containsKey(key)
    }
}
/**
 * Instantiates a new Default context map.
 */
